tests/ferret/core_test.go

```diff
@@ -533,6 +610,67 @@ func TestCore(t *testing.T) {
 				}},
 				IDs: []string{},
 			},
+
+			// binary
+
+			// $eq
+			{
+				name: "EqBinary",
+				q:    bson.D{{"value", bson.D{{"$eq", primitive.Binary{Subtype: 0x80, Data: []byte{42, 0, 13}}}}}},
+				IDs:  []string{"binary"},
+			},
+
+			// bool
+
+			// $eq
+
+			{
+				name: "EqBoolTrue",
+				q:    bson.D{{"value", bson.D{{"$eq", true}}}},
+				IDs:  []string{"bool-true"},
+			},
+			{
+				name: "EqBoolFalse",
+				q:    bson.D{{"value", bson.D{{"$eq", false}}}},
+				IDs:  []string{"bool-false"},
+			},
+
+			// datetime
+
+			// $eq
+
+			{
+				name: "EqDatetime",
+				q:    bson.D{{"value", bson.D{{"$eq", time.Date(2021, 11, 1, 10, 18, 42, 123000000, time.UTC)}}}},
+				IDs:  []string{"datetime"},
+			},
+
+			// timestamp
+
+			// $eq
+
+			{
+				name: "EqTimestamp",
+				q:    bson.D{{"value", bson.D{{"$eq", primitive.Timestamp{T: 42, I: 13}}}}},
+				IDs:  []string{"timestamp"},
+			},
+
+			// null
+
+			{
+				name: "EqNull",
+				q:    bson.D{{"value", bson.D{{"$eq", nil}}}},
+				o:    options.Find().SetSort(bson.D{{"_id", 1}}),
+				IDs:  []string{"array-three", "null"},
+			},
+
+			// regex
+
+			{
+				name: "FindRegex",
+				q:    bson.D{{"value", primitive.Regex{Pattern: "foo"}}},
```

Comment:
Let's add a test for `bson.D{{"value", bson.D{{ "$eq", primitive.Regex{Pattern: "foo"} }} }}`. My understanding is that should behave differently

Reply:
The test case has been added. Also additional case tests for regex have been added